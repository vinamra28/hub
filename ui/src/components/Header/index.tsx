import React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import {
  PageHeader,
  Brand,
  PageHeaderTools,
  Text,
  TextVariants,
  GridItem,
  Grid
} from '@patternfly/react-core';
import { useHistory } from 'react-router-dom';
import logo from '../../assets/logo/logo.png';
import Search from '../../containers/Search';

const Header: React.FC = () => {
  const history = useHistory();

  const headerTools = (
    <PageHeaderTools>
      <Grid>
        <GridItem span={11}>
          <Search />
        </GridItem>
      </Grid>
      <Text component={TextVariants.h3}>Login</Text>
    </PageHeaderTools>
  );

  return (
    <PageHeader
      logo={<Brand src={logo} alt="Tekton Hub Logo" onClick={() => history.push('/')} />}
      headerTools={headerTools}
    />
  );
};

export default Header;
