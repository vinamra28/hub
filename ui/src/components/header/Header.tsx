import React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import {
  PageHeader,
  Brand,
  PageHeaderToolsGroup,
  PageHeaderToolsItem,
  PageHeaderTools,
  Text,
  TextContent,
  TextVariants
} from '@patternfly/react-core';
import logo from '../../assets/logo/logo.png';
import SearchBar from '../searchbar/SearchBar';

const Header: React.FC = () => {
  const logoProps = {
    href: '/',
    target: ''
  };

  const headerTools = (
    <PageHeaderTools>
      <PageHeaderToolsGroup
        visibility={{
          default: 'visible',
          md: 'visible'
        }}
      >
        <PageHeaderToolsItem>
          <SearchBar />
        </PageHeaderToolsItem>

        <PageHeaderToolsItem>
          <TextContent>
            <Text component={TextVariants.h3}>Login</Text>
          </TextContent>
        </PageHeaderToolsItem>
      </PageHeaderToolsGroup>
    </PageHeaderTools>
  );

  return (
    <PageHeader
      logo={<Brand src={logo} alt="Tekton Hub Logo" />}
      headerTools={headerTools}
      logoProps={logoProps}
      showNavToggle
    />
  );
};

export default Header;
