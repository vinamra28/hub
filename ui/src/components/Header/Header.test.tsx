import React from 'react';
import renderer from 'react-test-renderer';
import Header from '.';

it('should render the header component', () => {
  const component = renderer.create(<Header />).toJSON();
  expect(component).toMatchSnapshot();
});
