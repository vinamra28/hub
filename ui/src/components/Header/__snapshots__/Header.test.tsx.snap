// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header component 1`] = `
<header
  className="pf-c-page__header"
>
  <div
    className="pf-c-page__header-brand"
  >
    <a
      className="pf-c-page__header-brand-link"
    >
      <img
        alt="Tekton Hub Logo"
        className="pf-c-brand"
        onClick={[Function]}
        src="logo.png"
      />
    </a>
  </div>
  <div
    className="pf-c-page__header-tools"
  >
    <div
      className="pf-l-grid"
    >
      <div
        className="pf-l-grid__item pf-m-11-col"
      >
        <input
          aria-invalid={false}
          aria-label="text input example"
          className="pf-c-form-control hub-search"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search for resources..."
          readOnly={false}
          required={false}
          spellCheck="false"
          type="search"
          value=""
        />
      </div>
    </div>
    <h3
      className=""
      data-pf-content={true}
    >
      Login
    </h3>
  </div>
</header>
`;
